[Merged by Bors] - chore(Fin/Basic): add 2 `@[simp]` attrs

urkud opened the pull request:
---
<!-- The text above the `---` will become the commit message when your
PR is merged. Please leave a blank newline before the `---`, otherwise
GitHub will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

To indicate co-authors, include lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

If you are moving or deleting declarations, please include these lines at the bottom of the commit message
(that is, before the `---`) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

eric-wieser commented:
Looks like this theorem is a duplicate, so I guess we should deprecate one of them.

urkud commented on Mathlib/Data/Fin/Basic.lean:
```diff
@@ -1058,7 +1060,7 @@ lemma succAbove_zero_apply (i : Fin n) : succAbove 0 i = succ i := by rw [succAb
 lemma succAbove_eq_last_iff {a : Fin (n + 2)} {b : Fin (n + 1)} (ha : a ≠ last _) :
     a.succAbove b = last _ ↔ b = last _ := by
-  simp [← succAbove_ne_last_last ha, succAbove_right_inj]
+  rw [← succAbove_ne_last_last ha, succAbove_right_inj]
```
UPD: explained here https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/.60simp.60.20cycle.20I.20fail.20to.20interpret/with/512895114

urkud commented:
> Looks like this theorem is a duplicate, so I guess we should deprecate one of them.

I've removed the relevant `@[simp]` change from this PR, awaiting Zulip.

b-mehta commented:
This looks sensible to me, @eric-wieser I'm not quite sure what you meant with the duplicate theorem, but I think deprecation of one can be in a separate PR, if you can find the duplication.

bors merge

urkud commented:
> This looks sensible to me, @eric-wieser I'm not quite sure what you meant with the duplicate theorem, but I think deprecation of one can be in a separate PR, if you can find the duplication.
> 
> bors merge

@b-mehta My actions caused confusion here: the first version of this PR was about a different lemma. It turned out that lemma has a `@[simp]` duplicate, so I removed `@[simp]` from that lemma and added `@[simp]` to the 2 lemmas in the current version of the PR. I should've opened a new PR instead.
